packages/legacy/core/App/utils/anonCredsProofRequestMapper.ts

```diff
@@ -0,0 +1,200 @@
+  AnonCredsProofRequestRestriction,
+  AnonCredsRequestedAttribute,
+  AnonCredsRequestedPredicate,
+  getAnonCredsTagsFromRecord,
+} from '@credo-ts/anoncreds'
+import {
+  DifPexCredentialsForRequest,
+  DifPresentationExchangeDefinition,
+  DifPresentationExchangeDefinitionV2,
+  W3cCredentialRecord,
+} from '@credo-ts/core'
+
+import { ProofCredentialAttributes, ProofCredentialPredicates } from '../types/proof-items'
+
+export type RecordWithMetadata = {
+  record: W3cCredentialRecord
+  anonCredsTags: AnonCredsCredentialTags
+}
+export type DescriptorMetadata = { [key: string]: RecordWithMetadata[] }
+
+export interface DifPexAnonCredsProofRequest extends AnonCredsProofRequest {
+  requested_attributes: Record<string, AnonCredsRequestedAttribute & { descriptorId?: string }>
+  requested_predicates: Record<string, AnonCredsRequestedPredicate & { descriptorId?: string }>
+}
+
+type FieldV2 = NonNullable<
+  DifPresentationExchangeDefinitionV2['input_descriptors'][number]['constraints']['fields']
+>[number]
+
+const getPredicateTypeAndValues = (predicateFilter: NonNullable<FieldV2['filter']>) => {
+  const predicates: {
+    predicateType: AnonCredsRequestedPredicate['p_type']
+    predicateValue: AnonCredsRequestedPredicate['p_value']
+  }[] = []
+
+  const supportedJsonSchemaNumericRangeProperties: Record<string, AnonCredsRequestedPredicate['p_type']> = {
+    exclusiveMinimum: '>',
+    exclusiveMaximum: '<',
+    minimum: '>=',
+    maximum: '<=',
+  }
+
+  for (const [key, value] of Object.entries(predicateFilter)) {
+    if (key === 'type') continue
+
+    const predicateType = supportedJsonSchemaNumericRangeProperties[key]
+    if (!predicateType) throw new Error(`Unsupported predicate filter property '${key}'`)
+    predicates.push({ predicateType, predicateValue: value })
+  }
+
+  return predicates
+}
+
+const getClaimNameForField = (field: FieldV2) => {
+  if (!field.path) throw new Error('Field path is required')
+  const baseClaimPath = '$.credentialSubject.'
+  const claimPaths = field.path.filter((path) => path.startsWith(baseClaimPath))
+  if (claimPaths.length === 0) return undefined
+
+  const claimNames = claimPaths.map((path) => path.slice(baseClaimPath.length))
+  const propertyName = claimNames[0]
+
+  return propertyName
+}
+
+export type ProcessedAttributes = { [key: string]: ProofCredentialAttributes }
+export type ProcessedPredicates = { [key: string]: ProofCredentialPredicates }
+
+export const createAnonCredsProofRequest = (
+  presentationDefinition: DifPresentationExchangeDefinition,
+  descriptorMetadata: DescriptorMetadata
+) => {
+  const anonCredsProofRequest: DifPexAnonCredsProofRequest = {
+    version: '1.0',
+    name: presentationDefinition.name ?? 'Proof request',
+    nonce: 'nonce',
+    requested_attributes: {},
+    requested_predicates: {},
+  }
+
+  const nonRevokedTime = Math.floor(Date.now() / 1000)
+  const nonRevokedInterval = { from: nonRevokedTime, to: nonRevokedTime }
```

Comment:
why shouldn`t these values be updated? 

[leaving out the `from` is semantically equivalent](https://github.com/hyperledger/aries-rfcs/blob/main/concepts/0441-present-proof-best-practices/README.md#semantics-of-non-revocation-interval-endpoints)

Reply:
okay my mistake about the from interval. Looking at it now I think the nonRevokedTime and nonRevokedInterval should be fine since they're created in the proof request object. I thought they were file-level variables for a second